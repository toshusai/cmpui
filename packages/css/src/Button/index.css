.cmpui_button {
  background-color: var(--cmpui-primary-color);

  align-items: center;
  border-radius: 9999px;
  border: none;
  box-sizing: border-box;
  color: white;
  cursor: pointer;
  display: flex;
  font-family: unset;
  font-size: 16px;
  height: 32px;
  justify-content: center;
  line-height: 16px;
  outline: none;
  padding: 0;
  padding: 8px 16px;
  text-align: center;
  text-decoration: none;
}

.cmpui_button:hover {
  background-color: var(--cmpui-primary-hover-color);
}

.cmpui_button:active {
  background-color: var(--cmpui-primary-active-color);
}

.cmpui_button:focus-visible {
  outline: var(--cmpui-primary-color) solid 1px;
  outline-offset: 1px;
}

/* disabled */

.cmpui_button:disabled {
  cursor: default;
  opacity: 0.5;
}

.cmpui_button:disabled:hover {
  background-color: var(--cmpui-primary-color);
}

.cmpui_button:disabled:active {
  background-color: var(--cmpui-primary-color);
}

/* Variant: secondary */

.cmpui_button[data-variant="secondary"] {
  background-color: var(--cmpui-secondary-color);
}

.cmpui_button[data-variant="secondary"]:hover {
  background-color: var(--cmpui-secondary-hover-color);
}

.cmpui_button[data-variant="secondary"]:active {
  background-color: var(--cmpui-secondary-active-color);
}

.cmpui_button[data-variant="secondary"]:disabled:hover {
  background-color: var(--cmpui-secondary-color);
}

.cmpui_button[data-variant="secondary"]:disabled:active {
  background-color: var(--cmpui-secondary-color);
}

/* Size: S */

.cmpui_button[data-size="S"] {
  font-size: 12px;
  height: 24px;
  line-height: 12px;
  padding: 4px 12px;
}
