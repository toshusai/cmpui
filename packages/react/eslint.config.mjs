"use strict";
import eslintConfigPrettier from "eslint-config-prettier";
import tseslint from "typescript-eslint";
import eslintPluginImport from "eslint-plugin-import";

export default [
  ...tseslint.configs.recommended,
  {
    ignores: ["dist/*", "node_modules/*", "storybook-static/*"],
  },
  {
    plugins: { import: eslintPluginImport },
    rules: {
      "vue/multi-word-component-names": "off",
      "no-unused-vars": "off",
      "@typescript-eslint/no-unused-vars": ["error"],
      "import/order": [
        "error",
        {
          "newlines-between": "always",
        },
      ],
    },
    languageOptions: {
      parserOptions: { parser: tseslint.parser, sourceType: "module" },
    },
  },
  eslintConfigPrettier,
];
