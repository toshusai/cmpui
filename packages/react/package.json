{
  "name": "@toshusai/cmpui",
  "version": "1.0.0-alpha.13",
  "description": "CmpUI is a component library designed for creating complex GUI applications in web browsers.",
  "main": "dist/index.mjs",
  "types": "dist/index.d.mts",
  "scripts": {
    "typecheck": "tsc -p tsconfig.ci.json",
    "test:watch": "vitest",
    "test": "vitest run",
    "dev": "tsup --watch",
    "build": "tsup",
    "storybook": "storybook dev -p 6007",
    "storybook:build": "storybook build",
    "storybook:test": "test-storybook",
    "storybook:test:ci": "concurrently -k -s first -n \"SB,TEST\" -c \"magenta,blue\" \"pnpm storybook:build && npx http-server storybook-static --port 6006 --silent\" \"wait-on http://127.0.0.1:6006 && pnpm storybook:test\"",
    "format": "prettier .",
    "format:fix": "prettier --write .",
    "lint": "eslint .",
    "lint:fix": "eslint --fix .",
    "fix": "npm run format:fix && npm run lint:fix"
  },
  "dependencies": {
    "@floating-ui/dom": "^1.6.10",
    "@radix-ui/react-context-menu": "^2.1.5",
    "@radix-ui/react-dropdown-menu": "^2.0.6",
    "@radix-ui/react-menubar": "^1.0.4",
    "@radix-ui/react-select": "^2.0.0",
    "@radix-ui/react-toast": "^1.1.5",
    "@toshusai/cmpui-core": "link:../core",
    "@toshusai/cmpui-css": "link:../css",
    "react-transition-group": "^4.4.5"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@playwright/test": "^1.41.2",
    "@storybook/addon-essentials": "8.2.6",
    "@storybook/addon-interactions": "8.2.6",
    "@storybook/addon-links": "8.2.6",
    "@storybook/addon-mdx-gfm": "8.2.6",
    "@storybook/addon-onboarding": "8.2.6",
    "@storybook/addon-themes": "8.2.6",
    "@storybook/blocks": "8.2.6",
    "@storybook/react": "8.2.6",
    "@storybook/react-vite": "8.2.6",
    "@storybook/test": "8.2.6",
    "@storybook/test-runner": "^0.17.0",
    "@storybook/theming": "8.2.6",
    "@tabler/icons-react": "^3.15.0",
    "@types/jest": "^29.5.11",
    "@types/react": "^18.2.37",
    "@types/react-dom": "^18.2.15",
    "@types/react-transition-group": "^4.4.11",
    "concurrently": "^8.2.2",
    "eslint": "^9.9.0",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-import": "^2.29.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "storybook": "8.2.6",
    "tsup": "^8.0.1",
    "typescript": "^5.2.2",
    "vitest": "^1.1.0",
    "wait-on": "^7.2.0"
  },
  "keywords": [
    "react",
    "typescript",
    "components",
    "ui",
    "gui"
  ],
  "author": "toshusai",
  "license": "MIT"
}
