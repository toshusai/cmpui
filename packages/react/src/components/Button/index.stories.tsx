import type { Meta, StoryObj } from "@storybook/react";

import { Button } from ".";

const meta: Meta<typeof Button> = {
  title: "Actions/Button",
  component: Button,
};

export default meta;
type Story = StoryObj<typeof Button>;

export const Basic: Story = {
  render: () => {
    return <Button>Button</Button>;
  },
};

export const Secondary: Story = {
  render: () => {
    return <Button variant="secondary">Button</Button>;
  },
};

export const Small: Story = {
  render: () => {
    return <Button size="S">Button</Button>;
  },
};

export const Disabled: Story = {
  render: () => {
    return <Button disabled>Button</Button>;
  },
};
