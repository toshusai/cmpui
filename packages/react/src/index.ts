export * from "./components/Button";
export * from "./components/CanvasView";
export * from "./components/Checkbox";
export * from "./components/Circle";
export * from "./components/ColorInput";
export * from "./components/ColorLoupe";
export * from "./components/ColorPickerCircle";
export * from "./components/ColorPickerRect";
export * from "./components/ColorSlider";
export * from "./components/ContextMenu";
export * from "./components/FloatBox";
export * from "./components/Grid";
export * from "./components/HueCircle";
export * from "./components/IconButton";
export * from "./components/IconButton/IconButtonGroup";
export * from "./components/Line";
export * from "./components/ListItem";
export * from "./components/MenuBar/MenuBar";
export * from "./components/MenuBar/MenuBarButton";
export * from "./components/Modal";
export * from "./components/Popover";
export * from "./components/RectGizmo";
export * from "./components/Ruler";
export * from "./components/Select";
export * from "./components/SelectRect";
export * from "./components/Slider";
export * from "./components/SliderNumberField";
export * from "./components/SplitPane";
export * from "./components/SVPicker";
export * from "./components/Switch";
export * from "./components/TextArea";
export * from "./components/TextInput";
export * from "./components/Toast";
export * from "./components/Tooltip";
export * from "./components/TreeView";
export * from "./utils";
