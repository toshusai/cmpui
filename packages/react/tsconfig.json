{
  "compilerOptions": {
    "target": "ES2020",
    "useDefineForClassFields": true,
    "lib": ["ES2020", "DOM", "DOM.Iterable"],
    "module": "ESNext",
    "skipLibCheck": true,

    "moduleResolution": "bundler",
    "allowImportingTsExtensions": true,
    "emitDeclarationOnly": true,
    "declaration": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "react-jsx",
    "outDir": "dist",

    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noFallthroughCasesInSwitch": true,
    "typeRoots": ["node_modules/@types", ".storybook/@types"]
  },
  "include": ["src"]
}
