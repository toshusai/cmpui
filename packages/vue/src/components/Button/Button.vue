<script setup lang="ts">
import { ButtonProps } from "./type";

withDefaults(defineProps<ButtonProps>(), {
  variant: "primary",
  size: "M",
  disabled: false,
});
</script>

<template>
  <button
    class="cmpui_button"
    :data-variant="variant"
    :data-size="size"
    :disabled="disabled"
  >
    <slot></slot>
  </button>
</template>
