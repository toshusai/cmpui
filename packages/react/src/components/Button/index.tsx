import { forwardRef } from "react";
import { CustomJSXElement } from "../../types/CustomJSXElement";
import { classNames } from "../../utils/classNames";

export type ButtonProps<T extends CustomJSXElement = "button"> = {
  as?: T;
  size?: "S" | "M";
  variant?: "primary" | "secondary";
} & Omit<React.ComponentProps<T>, "as">;

export const Button = forwardRef(function Button<T extends CustomJSXElement>(
  {
    variant = "primary",
    size = "M",
    children,
    className,
    as,
    ...props
  }: ButtonProps<T>,
  ref: React.ForwardedRef<HTMLButtonElement>,
) {
  const Component = as || "button";
  return (
    <Component
      ref={ref}
      type="button"
      className={classNames("cmpui_button", className)}
      {...props}
      data-variant={variant}
      data-size={size}
    >
      {children}
    </Component>
  );
}) as <T extends CustomJSXElement = "button">(p: ButtonProps<T>) => JSX.Element;
