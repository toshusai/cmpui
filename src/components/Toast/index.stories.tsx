import { Meta, StoryObj } from "@storybook/react";
import { useState } from "react";
import { Toast } from ".";
import { Button } from "../Button";

const meta: Meta<typeof Toast> = {
  title: "Components/Float/Toast",
  component: Toast,
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;

type Story = StoryObj<typeof Toast>;

export const Basic: Story = {
  render: function Render() {
    const [open, setOpen] = useState(false);

    return (
      <div
        style={{
          width: "100vw",
          height: 300,
          justifyContent: "center",
          alignItems: "center",
          display: "flex",
        }}
      >
        <Button onClick={() => setOpen(true)}>Open</Button>
        <Toast open={open} onOpenChange={setOpen}>
          <div
            style={{
              padding: 8,
            }}
          >
            Hello World
          </div>
        </Toast>
      </div>
    );
  },
};
