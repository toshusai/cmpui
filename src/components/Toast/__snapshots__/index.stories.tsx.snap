// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Float/Toast Basic smoke-test 1`] = `
<div style="width: 100vw; height: 300px; justify-content: center; align-items: center; display: flex;">
  <button type="button"
          class="cmpui_button"
          data-variant="primary"
          data-size="M"
  >
    Open
  </button>
  <div role="region"
       aria-label="Notifications (F8)"
       tabindex="-1"
       style="pointer-events: none;"
  >
    <ol tabindex="-1"
        class="cmpui_toast__viewport"
    >
    </ol>
  </div>
</div>
`;
